Uebersicht/UBAppDelegate.m

```diff
@@ -109,40 +113,69 @@ - (void)applicationDidFinishLaunching:(NSNotification *)aNotification
     [self listenToWallpaperChanges];
 }
 
-- (NSDictionary*)fetchState
+- (void)fetchState:(void (^)(NSDictionary*))callback
 {
-    [[UBWebSocket sharedSocket] open:[self serverUrl:@"ws"]];
+    [[UBWebSocket sharedSocket] open:[self serverUrl:@"ws"]
+                               token:token];
+
     NSURL *urlPath = [[self serverUrl:@"http"] URLByAppendingPathComponent: @"state/"];
-    NSData *jsonData = [NSData dataWithContentsOfURL:urlPath];
-    NSError *error = nil;
-    NSDictionary *dataDictionary = [NSJSONSerialization
-        JSONObjectWithData: jsonData
-        options: NSJSONReadingMutableContainers
-        error: &error
-    ];
-    if (error) NSLog(@"%@", error);
-    return dataDictionary;
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:urlPath];
+    [request setValue:@"Übersicht" forHTTPHeaderField:@"Origin"];
+    if (preferences.enableSecurity) {
```

Comment:
might seem like a small thing, but what do you think about having this setting only control whether the server checks the token. It can be sent in all other places regardless. 
This way the 'code surface' this setting impacts is reduced

Reply:
That's a good point!  I'll change the code to do that :)

Reply:
Updated.